tests/webTestServer.ts

```diff
@@ -640,31 +664,131 @@ function handleApiResolve(req: http.ServerRequest, res: http.ServerResponse) {
     });
 }
 
+function handleApiEnumerateTestFiles(req: http.ServerRequest, res: http.ServerResponse) {
+    readContent(req, (err, content) => {
+        try {
+            if (err) return sendInternalServerError(res, err);
+            if (!content) return sendBadRequest(res);
+            const tests: (string | FileTest)[] = enumerateTestFiles(content);
+            return sendJson(res, /*statusCode*/ 200, tests);
+        }
+        catch (e) {
+            return sendInternalServerError(res, e);
+        }
+    });
+}
+
+function enumerateTestFiles(runner: string) {
+    switch (runner) {
+        case "conformance":
+        case "compiler":
+            return listFiles(`tests/cases/${runner}`, /*serverDirname*/ undefined, /\.tsx?$/, { recursive: true }).map(parseCompilerTestConfigurations);
+        case "fourslash":

```

Comment:
For these other cases only the directory name varies -- could use a map or switch statement for just that, then `return listFiles(dir, ...);` unconditionally.

Reply:
True, but the code becomes more complicated and much less readable that way.